### PR TITLE
[IOTDB-2934] Fix StorageEngineV2 cannot recover

### PR DESCRIPTION
**Pull request by HTHou**

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/engine/StorageEngineV2.java [hunk 1]
```diff
@@ -103,6 +107,9 @@ public class StorageEngineV2 implements IService {
   private final ConcurrentHashMap<ConsensusGroupId, DataRegion> dataRegionMap =
       new ConcurrentHashMap<>();
 
+  /** number of ready virtual storage group processors */
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/engine/StorageEngineV2.java:110]:
```suggestion
  /** number of ready data region */
```

File: server/src/main/java/org/apache/iotdb/db/engine/StorageEngineV2.java [hunk 2]
```diff
@@ -229,44 +239,54 @@ public void recover() {
...
-    }
+  private void asyncRecover(ExecutorService pool, List<Future<Void>> futures) {
+
+    Map<String, List<String>> localDataRegionInfo = getLocalDataRegionInfo();
+    readyDataRegionNum = new AtomicInteger(0);
     // init wal recover manager
     WALRecoverManager.getInstance()
-        .setAllDataRegionScannedLatch(new CountDownLatch(sgCount * config.getDataRegionNum()));
+        .setAllDataRegionScannedLatch(
+            new CountDownLatch(localDataRegionInfo.size() * config.getDataRegionNum()));
```
Comment (HeimingZ) [inline server/src/main/java/org/apache/iotdb/db/engine/StorageEngineV2.java:249]: Only localDataRegionInfo.size() is OK

File: server/src/main/java/org/apache/iotdb/db/engine/StorageEngineV2.java [hunk 3]
```diff
@@ -229,44 +239,54 @@ public void recover() {
...
+            () -> {
+              DataRegion dataRegion = null;
+              try {
+                dataRegion = buildNewDataRegion(sgName, dataRegionId, Long.MAX_VALUE);
+              } catch (DataRegionException e) {
+                logger.error("Failed to recover data region {}[{}]", sgName, dataRegionId, e);
+              }
+              dataRegionMap.put(new DataRegionId(Integer.parseInt(dataRegionId)), dataRegion);
+              logger.info(
+                  "Data regions have been recovered {}", readyDataRegionNum.incrementAndGet());
```
Comment (HeimingZ) [inline server/src/main/java/org/apache/iotdb/db/engine/StorageEngineV2.java:263]: You can log readyDataRegionNum.incrementAndGet() / localDataRegionInfo.size() here.

File: server/src/main/java/org/apache/iotdb/db/engine/StorageEngineV2.java (hunk 2 above)
Comment (HTHou) [reply server/src/main/java/org/apache/iotdb/db/engine/StorageEngineV2.java]: localDataRegionInfo.size() is actually the sg number, but I have fixed the issue here.

File: server/src/main/java/org/apache/iotdb/db/engine/StorageEngineV2.java (hunk 3 above)
Comment (HTHou) [reply server/src/main/java/org/apache/iotdb/db/engine/StorageEngineV2.java]: It makes sense.

**Review by HeimingZ — APPROVED** [review]
LGTM

**Pull request merged**